typescript/lib/parsers/openai.ts

```diff
@@ -671,11 +684,17 @@ export class OpenAIChatModelParser extends ParameterizedModelParser<Chat.ChatCom
     const output = aiConfig.getLatestOutput(prompt);
     if (output != null) {
       if (output.output_type === "execute_result") {
-        const outputMessage =
-          output.data as unknown as Chat.ChatCompletionMessageParam;
         // If the prompt has output saved, add it to the messages array
-        if (outputMessage.role === "assistant") {
-          messages.push(outputMessage);
+        if (output.metadata?.role === "assistant") {
```

Comment:
I think we still need to handle the existing outputMessage-as-ChatCompletionMessageParam case here for backwards compatibility on deserialize. We can reserialize in our updated format, but it's possible the config being loaded has the old format

I think you can do similar check to above

```
if (typeof output.data === "string") {
   ... handle how you do here
} else if (output.data.hasOwnProperty("role")) {
  // for backwards compatibility
  const outputMessage = output.data as unknown as Chat.ChatCompletionMessageParam;
  if (outputMessage.role === "assistant") {
    messages.push(outputMessage);
  }
}

Reply:
You are correct, thank you for catching!